Update proxy documentation section

cywang117 opened the pull request:
Section should now reflect that the preferred method for enabling
redsocks is through the related Supervisor endpoints. Links have
been added. If the user prefers to not use the endpoints, the docs
for adding redsocks without touching the Supervisor API have also
been clarified.

Change-type: patch
Signed-off-by: Christina Wang <christina@balena.io>

cywang117 commented:
@balena-ci rebase

cywang117 commented:
@balena-ci I self-certify!

20k-ultra commented:
How did you find out all the info for this PR @cywang117. It contains a lot more than I had expected, nice!!

cywang117 commented:
@20k-ultra Thanks! Most of it was already there (meaning the suggested format when manually adding a .conf file), I added the parts that said we prefer to edit proxy configs through the Supervisor endpoint, and updated the recommended .conf fields to reflect what we set as unchangeable fields [here](https://github.com/balena-os/balena-supervisor/blob/master/src/host-config.ts#L19)
